core/src/main/scala/org/bitcoins/core/protocol/ln/LnHumanReadablePart.scala

```diff
@@ -8,87 +8,123 @@ import scodec.bits.ByteVector
+/** Prefix for genearting a LN invoice on the Bitcoin RegTest */
+case class lnbcrt(override val amount: Option[LnCurrencyUnit]) extends LnHumanReadablePart {
   def network: LnParams = LnBitcoinRegTest
-  def amount: Option[LnCurrencyUnit] = None
 }
 
 object LnHumanReadablePart {
-
-  private val allHRPs: Vector[LnHumanReadablePart] = Vector(lnbc, lntb, lnbcrt)
-
-  private def networkFromString(prefix: String): Try[LnParams] = {
-    val hrp = allHRPs.find(_.networkToString == prefix)
-    if (hrp.isDefined) { Success(hrp.get.network) }
-    else { Failure(new IllegalArgumentException(s"Unable to find a matching network, expected 'lnbc', 'lntb, or 'lnbcrt', got: $prefix")) }
-  }
+  private val lnbcNone = lnbc(None)
+  private val lntbNone = lntb(None)
+  private val lnbcrtNone = lnbcrt(None)
 
   def apply(network: NetworkParameters): Option[LnHumanReadablePart] = {
-    val lnNetwork = LnParams.fromNetworkParameters(network)
-    lnNetwork.map { params => HumanReadablePart(params, None) }
+    val lnNetworkOpt = LnParams.fromNetworkParameters(network)
+    lnNetworkOpt.map(LnHumanReadablePart.fromLnParams(_))
   }
 
   def apply(network: NetworkParameters, amount: LnCurrencyUnit): Option[LnHumanReadablePart] = {
-    val lnNetwork = LnParams.fromNetworkParameters(network)
-    lnNetwork.map { params => HumanReadablePart(params, Some(amount)) }
+    val lnNetworkOpt = LnParams.fromNetworkParameters(network)
+    lnNetworkOpt.map(ln => LnHumanReadablePart(ln, Some(amount)))
   }
 
   def apply(network: LnParams): LnHumanReadablePart = {
-    HumanReadablePart(network, None)
+    fromLnParams(network)
   }
 
-  def apply(network: LnParams, amount: LnCurrencyUnit): LnHumanReadablePart = {
-    HumanReadablePart(network, Some(amount))
+  /**
+   * Will return a [[org.bitcoins.core.protocol.ln.LnHumanReadablePart]]
+   * without a [[LnCurrencyUnit]] encoded in the invoice
+   * @param network
+   * @return
+   */
+  def fromLnParams(network: LnParams): LnHumanReadablePart = {
+    network match {
+      case LnParams.LnBitcoinMainNet => lnbcNone
+      case LnParams.LnBitcoinTestNet => lntbNone
+      case LnParams.LnBitcoinRegTest => lnbcrtNone
+    }
   }
 
-  def apply(input: String): Try[LnHumanReadablePart] = {
-    val networkPattern: Regex = "^[a-z]*".r //Select all of the letters, until we hit a number, as the network
-    val networkString = networkPattern.findFirstIn(input).getOrElse("")
-    val amountString = input.substring(networkString.length) //Select the remaining part of the string as the amount
-
-    val lnNetwork = networkFromString(networkString)
-    val amount = LnCurrencyUnits.fromEncodedString(amountString).toOption
+  /**
+   * Will return a [[org.bitcoins.core.protocol.ln.LnHumanReadablePart]]
+   * with the provide [[LnCurrencyUnit]] encoded in the invoice
+   * @param network
+   * @param amount
+   * @return
+   */
+  def apply(network: LnParams, amount: Option[LnCurrencyUnit]): LnHumanReadablePart = {
```

Comment:
What value are we getting out of defining `lnbc`, `lntb`, and `lnbcrt`? Using them leads to us have two matching statements for both `apply(LnParams)` and `apply(LnParams, Option[LnCurrencyUnit]`. The issue I see is that we already know if something is a `LnParam` type, its valid. What value are we getting my matching to `lnbc`, `lntb`, or`lnbcrt`?

Could we simply forego `lnbc`, `lntb`, and `lnbcrt`? That would remove the matching statements inside `apply(LnParams)` and replace them with `LnHumanReadablePart(LnParams, None)`.  

Similarly, the matching statement inside `apply(LnParams, Option[LnCurrencyUnit]` would be replaced`LnHumanReadablePart(LnParams, Some(amount))`.